app/components/tracks/TracksController.js

```diff
@@ -31,6 +33,57 @@ tracksModule.controller('TracksController',
                 }, function (error) {
                     console.error(error);
                 });
+
+            ApiJsonFactory.getJson('sessions')
+                .then(function (response) {
+                    tc.Sessions = response.data.sessions;
+                    $sessionStorage.sessions = tc.Sessions;
+                }, function (error) {
+                    console.error(error);
+                });
         }
 
-}]);
\ No newline at end of file
+        tc.showSession = function(track, event) {
+            trackThis = track;
+            $mdDialog.show({
+                controller: 'TrackDialogController',
+                templateUrl: 'app/components/tracks/trackdialog.html',
+                parent: angular.element(document.body),
+                targetEvent: event,
+
+            });
+        };
+
+
+}]);
+
+/*---------------------Dialog-----------------------*/
+
+tracksModule.controller('TrackDialogController', 
+    ['$mdDialog', '$sessionStorage',
+        function($mdDialog, $sessionStorage) {
+            var tdc = this;
+            tdc.track = trackThis;
+            tdc.sessions = $sessionStorage.sessions;
+            var count = 0;
+            for(var i = 0; i < tdc.sessions.length; i++) {
+                if(trackThis.id == tdc.sessions[i].track) {
+                    count++;                    
+                }
+                
+            }
+            tdc.count = count;
+            var k = 0;
+            tdc.tsessions = new Array(count);
+            for(var i = 0; i < tdc.sessions.length;i++) {
```

Comment:
Can you think of a way to have one helper function that could be used to count tracks and sessions? That would be nice. Also, I suggest using the object equality operator ```===```